tests/__init__.py

```diff
@@ -35,3 +42,41 @@ def unreadable_pdf(tmp_path: Path) -> str:
     file_path = tmp_path / "document.pdf"
     file_path.touch(mode=0o000)
     return str(file_path)
+
+
+def get_qt_app() -> Application:
+    if Application.instance() is None:
+        return Application()
+    else:
+        return Application.instance()
+
+
+def generate_isolated_updater(
+    tmp_path: Path, monkeypatch: MonkeyPatch, enable_qt: bool = False
+) -> Updater:
+    """Generate an Updater class with its own settings."""
+    if enable_qt:
+        app = get_qt_app()
+    else:
+        app = mock.MagicMock()
+
+    dummy = Dummy()
+    # XXX: We can monkey-patch global state without wrapping it in a context manager, or
+    # worrying that it will leak between tests, for two reasons:
+    #
+    # 1. Parallel tests in PyTest take place in different processes.
+    # 2. The monkeypatch fixture tears down the monkey-patch after each test ends.
+    monkeypatch.setattr(util, "get_config_dir", lambda: tmp_path)
```

Comment:
I'm probably missing something here. In this snippet, I'm describing that we don't need a teardown, due to the way Pytest runs the tests. Can you explain a bit more what you have in mind?

Reply:
See for example this:

```python
@pytest.fixture
def receiving_user(mail_admin):
    user = mail_admin.create_user()
    yield user
    mail_admin.delete_user(user)
```
That way, when when the fixture is loaded, it first runs a setup routine `mail_admin.create_user()`, then "generates" the user, which gets made use of in the fixture-calling function and only after that function is done, does it run the teardown routine `mail_admin.delete_user(user)`.

Essentially, it moves away setup and teardown code into the fixture rather than the test function.

What do you think?

Reply:
Hm, I'm still not in sync. I understand that if a fixture has teardown logic in the test where it's used, we can use `yield` to move that teardown logic within the fixture. The problem is that I don't think our tests have such a logic for this fixture. Am I missing something?

Reply:
And I think this way you could even remove monkeypatch from the previous test.

Reply:
I just realized one more issue why this can't work. The `tmp_path` value must be passed by the user to the `generate_isolated_updater` function, because there are cases where we want to control the settings path that this updater will use. If we convert this to a fixture, we lose this ability. With this in mind, I propose keeping it as is.